CMakeLists.txt

```diff
@@ -396,6 +397,10 @@ if(FORCE_ASSERT)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDUCKDB_FORCE_ASSERT")
 endif()
 
+if(MOLD_LINKER)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=mold")
```

Comment:
Is there a reason we're only using mold for executables?

Reply:
Shared library also use Mold now !  Thanks @Mause for pointing it out 👍
I think static libraries don't need linker flags because they are mostly archived by ar. (not entirely sure ?)